pkg/apis/apis/v1alpha1/types_apiexport.go

```diff
@@ -210,20 +209,28 @@ type PermissionClaim struct {
 	IdentityHash string `json:"identityHash,omitempty"`
 }
 
+// +kubebuilder:validation:XValidation:rule="has(self.__namespace__) || has(self.name)",message="at least one field must be set"
 type ResourceSelector struct {
 	// name of an object within a claimed group/resource.
 	// It matches the metadata.name field of the underlying object.
+	// If namespace is unset, all objects matching that name will be claimed.
```

Comment:
If ns is unset, wouldn't it be a cluster-scoped object? That's what I would expect from the rest of Kube.

Reply:
That's a very good question. Basically, yes, however in the authorizer we don't have any capability to check whether the request attributes reference a cluster-scoped object, hence my current implementation of the reverse permission authorizer doesn't make a difference and i don't see a similar one in RBAC authorizers from kube.

This is, however, somehow implied by the in-flight API Group/resource. But the invariant is not explicitely checked in the authorizer.

Conversely, one could set `name=foo` for a resource selector on a namespaced API Group/resource and the authorizer would reject any resource in any namespace matching that name.

This is how kube works as well. One can define a cluster role like:
```
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: cm-reader
rules:
- apiGroups: [""]
  resources: ["configmaps"]
  resourceNames: ["foo"]
  verbs: ["get", "watch", "list"]
```

and bind that to users. Then, a user will only be able to read configmap `foo` in any namespace but will be denied to modify any of them.